### ISSUE TITLE
MediaPlayer.STATE.COMPLETE is not resettable

### ISSUE THREAD

**Issue by LarsGrevelink** [bbc/tal#364]
Hi guys,

We are looking into the media players some more and are wondering why the `MediaPlayer.STATE.COMPLETE` is not resettable. Though it reached the end of a stream, and is therefore stopped, we need to transition it to a `MediaPlayer.STATE.STOPPED` state in order to reset it.

When using e.g. playlists, stopped has a total different outcome in comparison to complete. Since stopped means generally means a 'cancel' of playback and the end of stream events should mean playing the next media item. Calling `MediaPlayer.stop();` and triggering a stop event seems a bit obsolete.

We are planning to change this on our side. Would you like to have this change as well and what reasons did you guys have to not allow this in the first place? Did you encounter some devices to have issues with this?

Let me know and thanks in advance,

Lars Grevelink
24i

**Comment by LarsGrevelink** [comment]
BUMP :)

**Comment by tsadler1988** [comment]
Hi Lars,

We've discussed this a little internally, but haven't come to an agreed solution going forwards.

Here are my thoughts so far:

**tldr:** In my opinon, reset() should be callable from any state.

- From a functionality point of view, COMPLETE is equivalent to PAUSED (except resume isn't an option), so should PAUSED be resettable too?
- The differences between COMPLETE and STOPPED are that you cannot seek (playFrom) from STOPPED but you can from COMPLETE.
- Other differences are dependant on whether it's HTML5/CEHTML/Samsung Maple (which is where I'm starting to think STOPPED is not a well abstracted state) e.g.:
  - HTML5 in the STOPPED state, when previously in the COMPLETE state, will show the last frame of the video, whereas the other 2 will not (I think - not 100% sure)
- Our TAL apps never call just stop(), we always call stop() and then immediately reset().
- Having a brief look at our 3 primary media player strategies (HTML5, CEHTML, Samsung Maple), I don't think it would be an issue to call reset() from any state.

So, in my opinion, reset() should be callable from any state (sending you to the EMPTY state), and stop() should be deprecated. In the longer term, I would like to see stop() removed, and the STOPPED state renamed to SOURCE_SET or similar.

Looking forward to hearing feedback internally and externally on this.

Tom

**Comment by LarsGrevelink** [comment]
Hi Tom,

Sorry to drop out of the conversation. Thanks for your answer. We'll have a look at this internally over at ours as well. I will get back with you about our view on the matter. To me the steps above are looking good. Though a stop event looks still necessary to me in terms of different possible reactions from a UI standpoint.

L

**Comment by tsadler1988** [comment]
Thanks for the feedback Lars.

I don't fully understand the need for stop - should it not be equivalent to pause as per HTML5?

Tom

**Comment by kukulaka** [comment]
Closing issue due to[ deprecation](https://github.com/bbc/tal#deprecation-announcement-12122022).
